gufe/protocols/protocol_settings.py

```diff
@@ -0,0 +1,381 @@
+from typing import Optional, Dict, Union, List
+
+from pydantic import BaseModel, validator
+from openff.units import unit
+
+
+class SystemSettings(BaseModel):
```

Comment:
It will be critical to sync up with the Open Force Field Initiative folks to come up with a self-consistent division between:

1. Choices that impact the fully interacting potential energy function (which OpenFF is aiming to fully encode in the force field definition)
2. Choices that impact the free energy difference computed for a transformation (atom mapping, alchemical modification functional form, thermodynamic state at which the calculation is performed)
3. Choices that impact the efficiency of the calculation (integrators, barostats, collision rates, alchemical protocols, alchemical methods, alchemical intermediates, hydrogen mass)

Right now, you're mixing several of these together. This is highly problematic, and will lead directly to an inability to fulfill the primary mission of OpenFE (enabling free energy calculations with OpenFF force fields) because OpenFF is encoding all options that impact the fully interacting system potential function _within the force field definition_. Therefore, enabling options that impact the potential energy definition (such as `nonbonded_method`, `nonbonded_cutoff`, `constraints`, `rigid_water`) will mean you are failing to actually perform free energy calculations with the specified OpenFF force fields.

I'd be happy to help clarify these issues, but I think we need a design document that clearly separates these items.

Reply:
Yep, we had this discussion internally and had a good argument over whether e.g. the cutoff is free from the force field etc, there's room for improvement here.  

While we need to support openff force fields, I'm not sure we can support them at the exclusion of other force fields, so there might have to be variations on these settings where the parameters are stated separately (such as is happening here).

Reply:
There is nothing in my argument that says you should support OpenFF to the exclusion of other force fields.
My only point is that you _must_ support OpenFF force fields, which means that you will need to bundle all of the parameters that impact the potential energy for a fully interacting system together, separated from other parameters that impact simulation efficiency.

One way to do this would be to recognize that an OpenFF force field will bundle all of the parameters that impact the potential energy of a fully-interacting system into a single object. For other force fields, you can use a container object that similarly bundles everything together, explicitly specifying all the parameters left undefined by the choice of force field. These objects can then be used interchangeably to specify which "complete" force field should be used, and the choices of auxiliary parameters for traditional force fields like GAFF can subsequently be optimized to define best practices.

Reply:
> Yep, we had this discussion internally and had a good argument over whether e.g. the cutoff is free from the force field etc, there's room for improvement here.

Might I also suggest that internal discussions about how to represent force fields should likely also include a representative from OpenFF? There seems to be a great deal of accumulated knowledge within OpenFF that is not making its way into these discussions at the moment.

Reply:
> There is nothing in my argument that says you should support OpenFF to the exclusion of other force fields. My only point is that you _must_ support OpenFF force fields, which means that you will need to bundle all of the parameters that impact the potential energy for a fully interacting system together, separated from other parameters that impact simulation efficiency.
> 

Yep sorry misread this a bit at first.

Reply:
Another great cross-cutting discussion to have might be "which force fields should be supported?" and "how can we unify force field support across simulation engines?". There are a number of tools that have been developed that attempt to band-aid this situation for now, like [simtk.openmm.app.ForceField](http://docs.openmm.org/latest/userguide/application/02_running_sims.html#a-first-example), [`openmmforcefields.generators.SystemGenerator`](https://github.com/openmm/openmmforcefields#automating-force-field-management-with-systemgenerator), [OpenFF Interchange](https://openff-interchange.readthedocs.io/), and now [espaloma](http://github.com/choderalab/espaloma). If we know what you'd like to support, there's likely some things that can be done to make this easy that OpenFE need not solely support.